classes/abstracts/ActionScheduler_WPCLI_Command.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/**
+ * Abstract for WP-CLI commands.
+ */
+abstract class ActionScheduler_WPCLI_Command extends \WP_CLI_Command {
+
+	const DATE_FORMAT = 'Y-m-d H:i:s O';
+
+	/** @var string[] */
+	protected $args;
+
+	/** @var array<string, string> */
+	protected $assoc_args;
+
+	/**
+	 * Construct.
+	 *
+	 * @param string[]              $args       Positional arguments.
+	 * @param array<string, string> $assoc_args Keyed arguments.
+	 */
+	public function __construct( array $args, array $assoc_args ) {
+		if ( ! defined( 'WP_CLI' ) || ! constant( 'WP_CLI' ) ) {
+			/* translators: %s php class name */
+			throw new Exception( sprintf( __( 'The %s class can only be run within WP CLI.', 'action-scheduler' ), __CLASS__ ) );
+		}
+
+		$this->args       = $args;
+		$this->assoc_args = $assoc_args;
+	}
+
+	/**
+	 * Execute command.
+	 */
+	abstract public function execute();
+
+	/**
+	 * Get the scheduled date in a human friendly format.
+	 *
+	 * @see ActionScheduler_ListTable::get_schedule_display_string()
+	 * @param ActionScheduler_Schedule $schedule Schedule.
+	 * @return string
+	 */
+	protected function get_schedule_display_string( ActionScheduler_Schedule $schedule ) {
+
+		$schedule_display_string = '';
+
+		if ( ! $schedule->get_date() ) {
+			return '0000-00-00 00:00:00';
+		}
+
+		$next_timestamp = $schedule->get_date()->getTimestamp();
+
+		$schedule_display_string .= $schedule->get_date()->format( static::DATE_FORMAT );
+
+		return $schedule_display_string;
+	}
+
+	/**
+	 * Returns the recurrence of an action or 'Non-repeating'. The output is human readable.
+	 *
+	 * @see \ActionScheduler_ListTable::get_recurrence()
+	 * @param ActionScheduler_Action $action Action.
+	 *
+	 * @return string
+	 */
+	protected function get_recurrence( $action ) {
+		$schedule = $action->get_schedule();
+		if ( $schedule->is_recurring() ) {
+			$recurrence = $schedule->get_recurrence();
+
+			if ( is_numeric( $recurrence ) ) {
+				/* translators: %s: time interval */
+				return sprintf( __( 'Every %s', 'action-scheduler' ), self::human_interval( $recurrence ) );
+			} else {
+				return $recurrence;
+			}
+		}
+
+		return __( 'Non-repeating', 'action-scheduler' );
+	}
+
+	/**
+	 * Transforms arguments with '__' from CSV into expected arrays.
+	 *
+	 * @see \WP_CLI\CommandWithDBObject::process_csv_arguments_to_arrays()
+	 * @link https://github.com/wp-cli/entity-command/blob/6e0e77a297eefa3329b94bec16c15cf7528d343f/src/WP_CLI/CommandWithDBObject.php
+	 * @return void
+	 */
+	protected function process_csv_arguments_to_arrays() {
```

Comment:
When do we invoke this? I see it is called from `ActionScheduler_WPCLI_Action_List_Command::execute()`, to be clear, but can you give me an example of a command invocation that requires it?

Reply:
I cannot give an example, I only noticed that [several of the core commands](https://github.com/search?q=repo%3Awp-cli%2Fentity-command%20process_csv_arguments_to_arrays&type=code) implement it.

Reply:
OK, but if in our case we aren't accepting CSV-formatted arguments, why add it?

Reply:
Core's documentation doesn't mention accepting CSV-formatted arguments either, so it's likely they've anticipated some situation that should be accounted for.

If you really want to get rid of it, please do so.